File: app/src/main/java/de/bikebean/app/db/settings/settings/location_settings/Acc.java
package de.bikebean.app.db.settings.settings.location_settings;

import androidx.annotation.NonNull;

import de.bikebean.app.db.settings.settings.LocationSetting;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.state.State;

public class Acc extends LocationSetting {

    private static final @NonNull State.KEY key = State.KEY.ACC;

    public Acc(double acc, final @NonNull Sms sms) {
        super(acc, sms, key);
    }

    public Acc() {
        super(key);
    }
}

File: app/src/main/java/de/bikebean/app/db/settings/settings/add_to_list_settings/location_settings/Acc.kt
package de.bikebean.app.db.settings.settings.add_to_list_settings.location_settings

import de.bikebean.app.db.settings.settings.add_to_list_settings.LocationSetting
import de.bikebean.app.db.settings.settings.add_to_list_settings.WappState
import de.bikebean.app.db.state.State.KEY
import de.bikebean.app.ui.drawer.status.location.ResponseBody

class Acc : LocationSetting {

    constructor(
            responseBody: ResponseBody,
            wappState: WappState
    ) : super(responseBody.accuracy.toDouble(), wappState, key)

    constructor() : super(key)

    companion object {
        private val key = KEY.ACC
    }

}